boa/src/builtins/value/mod.rs

```diff
@@ -483,56 +486,52 @@ impl Value {
         let _timer = BoaProfiler::global().start_event("Value::update_property", "value");
 
         if let Some(ref mut object) = self.as_object_mut() {
-            // Use value, or walk up the prototype chain
-            if let Some(property) = object.properties_mut().get_mut(field) {
-                *property = new_property;
-            }
+            object.insert_property(field, new_property);
         }
     }
 
     /// Resolve the property in the object and get its value, or undefined if this is not an object or the field doesn't exist
     /// get_field recieves a Property from get_prop(). It should then return the [[Get]] result value if that's set, otherwise fall back to [[Value]]
     /// TODO: this function should use the get Value if its set
-    pub fn get_field<F>(&self, field: F) -> Self
+    pub fn get_field<Key>(&self, key: Key) -> Self
```

Comment:
As in other cases, using `Key` instead of `K` could look like we were using an actual type instead of a generic.